spotlight/evaluation.py

```diff
@@ -96,6 +96,60 @@ def sequence_mrr_score(model, test):
     return np.array(mrrs)
 
 
+def precision_recall_at_k(model, test, k):
+    """
+    Compute Precision@k and Recall@k scores. One score
+    is given for every user with interactions in the test
+    set, representing the Precision@k and Recall@k of all their
+    test items.
+
+    Parameters
+    ----------
+
+    model: fitted instance of a recommender model
+        The model to evaluate.
+    test: :class:`spotlight.interactions.Interactions`
+        Test interactions.
+    k: int or array of int,
+        The maximum number of predicted items
+    Returns
+    -------
+
+    (Precision@k, Recall@k): numpy array of shape (num_users,)
+        A tuple of Precisions@k and Recalls@k for each user in test.
+    """
+
+    test = test.tocsr()
+
+    if not isinstance(k, list):
+        k = [k]
+
+    precisions = []
+    recalls = []
+
+    for user_id, row in enumerate(test):
+
+        if not len(row.indices):
+            continue
+
+        precision_at_k = []
+        recall_at_k = []
+
+        predictions = -model.predict(user_id)
+        predictions = predictions.argsort()
+        targets = row.indices
+
+        for _k in k:
+            pred = np.asarray(predictions)[:_k]
```

Comment:
Minor: predictions is already a numpy array.

Reply:
Yes, you are correct.